### DISCUSSION TITLE
How to decrypt message content?

### DISCUSSION THREAD

**Discussion by Coooz** [nextgenhealthcare/connect#4821] [Q&A] [answered]
Hi,

I checked _Encrypt message content_ in a channel's Message Storage settings on the Summary tab. And indeed, the message content gets encrypted in the database.
The Encrypt message content tooltip mentions that "the content will not be searchable". But what if I regret this setting and I actually do want to search the database? Is there a way to decrypt the contents?

Thanks in advance.

**Comment by pacmano1** [comment]
I believe this is answered at https://forums.mirthproject.io/forum/mirth-connect/support/7578-how-to-decrypt-encrypted-messages-stored-in-message-table?p=63274#post63274

Encrypting messages make it very painful to use Mirth IMHO.

**Reply by Coooz** [reply]
Hi pacmano1,

I believe the post you suggested is somewhat outdated... it contains getHL7Serializer() which got me a "deprecated" error, and SQL FROM always expects a database table as far as I know. "message" here got me an error as well.
That said, the post contains enough useful information to help me along.

All that's needed is the following Database Reader SQL:...
```
SELECT CONTENT
FROM D_MC24 -- channel 24 contains encrypted messages
WHERE IS_ENCRYPTED = '1' AND DATA_TYPE = 'HL7V2'
```
...and this JavaScript Source transformer:
```
var encryptor = com.mirth.connect.server.controllers.ControllerFactory.getFactory().createConfigurationController().getEncryptor();
var decrypted = encryptor.decrypt(msg['content']).toString();
tmp = SerializerFactory.getSerializer('HL7V2').toXML(decrypted);
```
And that actually does the trick. Thank you for helping me out here!

**Reply by pacmano1** [reply]
Great info.  I've never encrypted the message store actually, so I didn't try that forum post myself.  Since Nick wrote it I just assumed it was correct.

**Comment by tonygermano** [comment, marked as answer]
_Solution copied from thread:_

All that's needed is the following Database Reader SQL:...

```sql
SELECT CONTENT
FROM D_MC24 -- channel 24 contains encrypted messages
WHERE IS_ENCRYPTED = '1' AND DATA_TYPE = 'HL7V2'
```

...and this JavaScript Source transformer:

```js
var encryptor = com.mirth.connect.server.controllers.ControllerFactory.getFactory().createConfigurationController().getEncryptor();
var decrypted = encryptor.decrypt(msg['content']).toString();
tmp = SerializerFactory.getSerializer('HL7V2').toXML(decrypted);
```
